docs/java/usage.asciidoc

```diff
@@ -0,0 +1,218 @@
+== Getting started
+
+=== Maven Repository
+
+The low-level Java REST client is hosted on
+http://search.maven.org/#search%7Cga%7C1%7Cg%3A%22org.elasticsearch.client%22[Maven
+Central]. The minimum Java version required is `1.7`.
+
+Here is how you can configure the dependency using maven as a dependency manager.
+Add the following to your `pom.xml` file:
+
+[source,xml]
+--------------------------------------------------
+<dependency>
+    <groupId>org.elasticsearch.client</groupId>
+    <artifactId>rest</artifactId>
+    <version>${es.version}</version>
+</dependency>
+--------------------------------------------------
+
+=== Dependencies
+
+The low-level Java REST client internally uses the
+http://hc.apache.org/httpcomponents-asyncclient-dev/[Apache Http Async Client]
+ to send http requests. It depends on the following artifacts, namely the async
+ http client and its own transitive dependencies:
+
+- org.apache.httpcomponents:httpasyncclient
+- org.apache.httpcomponents:httpcore-nio
+- org.apache.httpcomponents:httpclient
+- org.apache.httpcomponents:httpcore
+- commons-codec:commons-codec
+- commons-logging:commons-logging
+
+
+=== Initialization
+
+A `RestClient` instance can be built through the corresponding
+`RestClientBuilder` class, created via `RestClient#builder(HttpHost...)`
+static method. The only required argument is one or more hosts that the
+client will communicate with, provided as instances of
+https://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/HttpHost.html[HttpHost]
+ as follows:
+
+[source,java]
+--------------------------------------------------
+RestClient restClient = RestClient.builder(
+        new HttpHost("localhost", 9200),
+        new HttpHost("localhost", 9201)).build();
+--------------------------------------------------
+
+The `RestClient` class is thread-safe and ideally has the same lifecycle as
+the application that uses it. It is important that it gets closed when no
+longer needed so that all the resources used by it get properly released,
+as well as the underlying http client instance and its threads:
+
+[source,java]
+--------------------------------------------------
+restClient.close();
+--------------------------------------------------
+
+`RestClientBuilder` also allows to optionally set the following configuration
+parameters while building the `RestClient` instance:
+
+- `setDefaultHeaders`: default headers that need to be sent with each request,
```

Comment:
yes it would, if we had them published somewhere, I don't want to rely on external services. I have asked internally to move forward with javadocs publishing, hopefully we will have this soon